class/abstract-hd-segwit-p2sh-wallet.js

```diff
@@ -180,6 +180,36 @@ export class AbstractHDSegwitP2SHWallet extends AbstractHDWallet {
     return this._xpub;
   }
 
+  async getXpriv() {
+    if (this._xpriv) {
+      return this._xpriv; // cache hit
+    }
+    // first, getting xpub
+    this.seed = await this.getSeed();
+    const root = HDNode.fromSeed(this.seed, config.network);
+    const path = this._getPath();
+    const child = root.derivePath(path);
```

Comment:
since we are reusing the same for getKeyPair maybe we could make it a function?

Reply:
moved reusable code to separate method